src/js/route.js

```diff
@@ -1,267 +1,218 @@
-import UrlBuilder from './UrlBuilder';
-import { stringify } from 'qs';
+import { parse, stringify } from 'qs';
 
-class Router extends String {
-    constructor(name, params, absolute, customZiggy = null) {
-        super();
+class Route {
+    absolute = true;
+    bindings = {};
 
-        this.name = name;
-        this.absolute = absolute;
-        this.ziggy = customZiggy ? customZiggy : Ziggy;
-        this.urlBuilder = this.name ? new UrlBuilder(name, absolute, this.ziggy) : null;
-        this.template = this.urlBuilder ? this.urlBuilder.construct() : '';
-        this.urlParams = this.normalizeParams(params);
-        this.queryParams = {};
-        this.hydrated = '';
+    constructor(name, definition, config) {
+        Object.entries({ name, ...definition, ...config }).map(([key, value]) => this[key] = value);
```

Comment:
Let's instead store these as `name` and `definition` and `config` props so it's clearer where each property you're calling later comes from